### ISSUE TITLE
Can't run yubioath-flutter 7.0.0 on Ubuntu 24.04 LTS

### ISSUE THREAD

**Issue by jacklund** [Yubico/yubioath-flutter#1578]
Downloaded the 7.0.0-linux tarball from your website. When I unpack it and run `./authenticator`, I get:

```
% ./authenticator
libEGL warning: MESA-LOADER: failed to open radeonsi: libLLVM-15.so.1: cannot open shared object file: No such file or directory (search paths /snap/alacritty/125/usr/lib/x86_64-linux-gnu/dri, suffix _dri)

libEGL warning: MESA-LOADER: failed to open zink: libLLVM-15.so.1: cannot open shared object file: No such file or directory (search paths /snap/alacritty/125/usr/lib/x86_64-linux-gnu/dri, suffix _dri)

libEGL warning: MESA-LOADER: failed to open swrast: libLLVM-15.so.1: cannot open shared object file: No such file or directory (search paths /snap/alacritty/125/usr/lib/x86_64-linux-gnu/dri, suffix _dri)


** (authenticator:13246): WARNING **: 16:25:22.207: Failed to start Flutter renderer: No GL implementation is available
```

If I install libllvm15t64, and then try again, I get:

```
% ./authenticator
libEGL fatal: DRI driver not from this Mesa build ('23.2.1-1ubuntu3.1~22.04.2' vs '24.0.5-1ubuntu1')
```

**Comment by AdamVe** [comment]
Hi, thank you for the report. Could you provide more details about your installation/system? When I tried to reproduce the issue with freshly installed and updated Ubuntu 24.04 LTS the app works as expected.

**Comment by AdamVe** [comment]
Could the issue be related to conflicting mesa packages installed on your system? I still cannot reproduce the issue, but found following askubuntu question which has a similar error message to yours. Could you take a look?
https://askubuntu.com/questions/1506303/issues-with-mesa-driver-after-update

**Comment by jacklund** [comment]
Hrm, didn't work. `ppa-purge` didn't work for me, so I tried the second one:

```
jacks-laptop:~/Downloads/yubico-authenticator-7.0.0-linux% sudo dpkg --force-all -P libgl1-mesa-dri:i386 libgl1-mesa-dri:amd64 mesa-opencl-icd:amd64
dpkg: warning: ignoring request to remove mesa-opencl-icd which isn't installed
dpkg: libgl1-mesa-dri:amd64: dependency problems, but removing anyway as you requested:
 xorg depends on libgl1-mesa-dri.
 steam-libs:amd64 depends on libgl1-mesa-dri.
 libglx-mesa0:amd64 depends on libgl1-mesa-dri.

(Reading database ... 178564 files and directories currently installed.)
Removing libgl1-mesa-dri:amd64 (24.0.5-1ubuntu1) ...
Purging configuration files for libgl1-mesa-dri:amd64 (24.0.5-1ubuntu1) ...
dpkg: libgl1-mesa-dri:i386: dependency problems, but removing anyway as you requested:
 steam-libs:i386 depends on libgl1-mesa-dri.
 libglx-mesa0:i386 depends on libgl1-mesa-dri.
Removing libgl1-mesa-dri:i386 (24.0.5-1ubuntu1) ...
Purging configuration files for libgl1-mesa-dri:i386 (24.0.5-1ubuntu1) ...

jacks-laptop:~/Downloads/yubico-authenticator-7.0.0-linux% sudo apt install -f
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Correcting dependencies... Done
The following packages were automatically installed and are no longer required:
  attr libcephfs2 samba-ad-provision samba-vfs-modules tdb-tools
Use 'sudo apt autoremove' to remove them.
The following additional packages will be installed:
  libgl1-mesa-dri libgl1-mesa-dri:i386
Recommended packages:
  libgl1-amber-dri:i386
The following NEW packages will be installed:
  libgl1-mesa-dri libgl1-mesa-dri:i386
0 upgraded, 2 newly installed, 0 to remove and 34 not upgraded.
Need to get 18.1 MB of archives.
After this operation, 70.4 MB of additional disk space will be used.
Do you want to continue? [Y/n] y
Get:1 http://us.archive.ubuntu.com/ubuntu noble/main amd64 libgl1-mesa-dri amd64 24.0.5-1ubuntu1 [8939 kB]
Get:2 http://us.archive.ubuntu.com/ubuntu noble/main i386 libgl1-mesa-dri i386 24.0.5-1ubuntu1 [9200 kB]
Fetched 18.1 MB in 1s (22.5 MB/s)
Selecting previously unselected package libgl1-mesa-dri:amd64.
(Reading database ... 178532 files and directories currently installed.)
Preparing to unpack .../libgl1-mesa-dri_24.0.5-1ubuntu1_amd64.deb ...
Unpacking libgl1-mesa-dri:amd64 (24.0.5-1ubuntu1) ...
Selecting previously unselected package libgl1-mesa-dri:i386.
Preparing to unpack .../libgl1-mesa-dri_24.0.5-1ubuntu1_i386.deb ...
Unpacking libgl1-mesa-dri:i386 (24.0.5-1ubuntu1) ...
Setting up libgl1-mesa-dri:amd64 (24.0.5-1ubuntu1) ...
Setting up libgl1-mesa-dri:i386 (24.0.5-1ubuntu1) ...

jacks-laptop:~/Downloads/yubico-authenticator-7.0.0-linux% ./authenticator
libEGL fatal: DRI driver not from this Mesa build ('23.2.1-1ubuntu3.1~22.04.2' vs '24.0.5-1ubuntu1')
```

**Comment by AdamVe** [comment]
Thanks for trying and reporting back. I continue investigations here, still I was not able to reproduce the issue so if you have time and could try the following it might help us to track down the issue.
- Could you try to run the authenticator from your command line as `LIBGL_ALWAYS_SOFTWARE=1 ./authenticator` instead of just `./authenticator`?
- Do you remember if you did any specific upgrades of the system/drivers? For example: originally the system was 20.04 LTS, then it was upgraded to 22.04 and later to 24.04 LTS.
- Could you try to install any of the older 6.x releases from our release page https://developers.yubico.com/yubioath-flutter/Releases/ (or from GitHub)? Does any of them work?

**Comment by jacklund** [comment]
> Thanks for trying and reporting back. I continue investigations here, still I was not able to reproduce the issue so if you have time and could try the following it might help us to track down the issue.
> 
>     * Could you try to run the authenticator from your command line as `LIBGL_ALWAYS_SOFTWARE=1 ./authenticator` instead of just `./authenticator`?

```
% LIBGL_ALWAYS_SOFTWARE=1 ./authenticator
libEGL fatal: DRI driver not from this Mesa build ('23.2.1-1ubuntu3.1~22.04.2' vs '24.0.5-1ubuntu1')
```

>     * Do you remember if you did any specific upgrades of the system/drivers? For example: originally the system was 20.04 LTS, then it was upgraded to 22.04 and later to 24.04 LTS.

So, it _did_ come originally with 22.04, and I upgraded it to 24.04 soon after getting it, because there were some things that weren't working quite right, and  I thought upgrading would help (it did). Also, full disclosure, I just like to be on the latest codebase.
> 
>     * Could you try to install any of the older 6.x releases from our release page https://developers.yubico.com/yubioath-flutter/Releases/ (or from GitHub)? Does any of them work?

Tried a few, no luck, here's a representative:

```
jacks-laptop:~/Downloads/yubico-authenticator-6.1.0-linux% ./authenticator
libEGL fatal: DRI driver not from this Mesa build ('23.2.1-1ubuntu3.1~22.04.2' vs '24.0.5-1ubuntu1')
```

Honestly, it could be just that this thing has some specialized hardware on it from Framework. The kernel is an OEM kernel, so there could be something going on with device drivers for this thing (I don't know if you're familiar with Framework, they make laptops you can service yourself, so they have some specialized hardware, I imagine). 

One thing I might try, if I get some time, is installing 24.04 from scratch rather than as an upgrade, see if that makes a difference.

Thanks for all the time you're putting into this, I really appreciate it.

-Jack

**Comment by jacklund** [comment]
Okay, figured it out.

After re-reading [this StackOverflow article](https://askubuntu.com/questions/1504501/libgl-error-dri-driver-not-from-this-mesa-build-23-0-4-0ubuntu122-04-1-vs), I noticed the comment near the bottom about using the wrong terminal emulator, and instead using the integrated one. I use Alacritty, so I tried the basic `terminal` one.

And it worked. :facepalm: 

TIL that the different terminal emulators engage different GL drivers. Gah!

Sorry for all the wasted time. What I'll probably end up doing is making a desktop entry for this so that I can just access it through the launcher.

Apologies again for the confusion.

-Jack
